external/llvm-project/mlir/lib/Dialect/Tosa/Transforms/TosaPartition.cpp

```diff
@@ -321,21 +322,66 @@ void outlineConvPartOps(Operation *convOp, ArrayRef<Operation *> secondOps,
 
   // Erase the ops we outlined, which should be safe now.
   for (auto &op : llvm::make_early_inc_range(llvm::reverse(secondOps))) {
-    assert(op->use_empty() && "expected 'op' to have no uses");
-    op->erase();
+    if (op->use_empty()) {
+      op->erase();
+    }
   }
   assert(convOp->use_empty() && "expected 'op' to have no uses");
   convOp->erase();
   for (auto &op : llvm::make_early_inc_range(frontOps)) {
-    assert(op->use_empty() && "expected 'op' to have no uses");
-    op->erase();
+    if (op->use_empty()) {
+      op->erase();
+    }
   }
 }
 
 // Inspired by / adapted from TestSCFIfUtilsPass in
 // test/lib/Transforms/TestSCFUtils.cpp.
 class TosaPartitionPass : public TosaPartitionBase<TosaPartitionPass> {
 public:
+  bool isConstantZero(Operation *op) {
+    // test for zero
```

Comment:
Can this be simplified using llvm::all_of(), as in isZeroAttribute() in external/llvm-project/mlir/lib/Dialect/LLVMIR/IR/LLVMDialect.cpp?

Reply:
Good call, I will look into that.